script/eslint-plugin-va/rules/prefer-web-component-library.js

```diff
@@ -0,0 +1,98 @@
+const MESSAGE =
+  '<{{ reactComponent }}> can be replaced by <{{ webComponent }}>';
+
+const getPropNode = (node, propName) =>
+  node.openingElement.attributes.find(n => n.name.name === propName);
+
+// Allows us to check if the JSX element we've identified is imported from
+// the component library. Some applications have their own components with the
+// same names as Design System components.
+// TODO: Consider improving the performance of this w/ memoization
+const isLibraryImport = (context, componentName) => {
+  let isLibraryComponent = false;
+  const allImports = context
+    .getAncestors()[0]
+    .body.filter(node => node.type === 'ImportDeclaration');
+  const componentLibraryImports = allImports.filter(node =>
+    node.source.value.includes(
+      '@department-of-veterans-affairs/component-library',
+    ),
+  );
+
+  componentLibraryImports.forEach(imp => {
+    if (imp.specifiers.find(i => i.local.name === componentName))
+      isLibraryComponent = true;
+  });
+
+  return isLibraryComponent;
+};
```

Comment:
The performance on this probably isn't great considering we are currently running it on each JSX element in the syntax tree. The main goal here is to build migration functionality for `<Telephone>` - I think we can make improvements to the rule if we continue to develop it.

This is also less applicable for `<Telephone>` specifically, but I was doing this with `<AlertBox>` in mind, and there are a few others that fall into that category as well.

Reply:
yes, if there are ways to improve the performance of this, that would be nice. If it is not applicable for `<Telephone>` don't run it for now and maybe with the next iteration we can improve it.